src/ring_theory/polynomial/hermite.lean

```diff
@@ -0,0 +1,67 @@
+/-
+Copyright (c) 2023 Luke Mantle. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Luke Mantle
+-/
+
+import data.polynomial.derivative
+
+/-!
+# Hermite polynomials
+
+This file defines `hermite n`, the nth probabilist's Hermite polynomial.
+
+## Main definitions
+
+* `x_sub_dx`  : the operation `(x - d/dx)` used to recursively define the Hermite polynomials
+* `hermite n` : the nth probabilist's Hermite polynomial, defined as a `polynomial ℤ`, using
+                `x_sub_dx` recursively
+
+## References
+
+[Hermite Polynomials](https://en.wikipedia.org/wiki/Hermite_polynomials)
+
+-/
+
+noncomputable theory
+open polynomial
+
+section x_sub_dx
+
+variables {R : Type} [comm_ring R] (p : polynomial R)
+
+/-- the operation `(x - d/dx)` defined for `polynomial R` -/
+def x_sub_dx := X * p - p.derivative
+
+@[simp] lemma x_sub_dx_def : x_sub_dx p = X * p - p.derivative := rfl
+
+lemma x_sub_dx_apply {x : R} :
+  eval x (x_sub_dx p) = x * (eval x p) - (eval x (derivative p)) := by simp
+
+@[simp] lemma x_sub_dx_map {S : Type} [comm_ring S] {f : R →+* S} :
+  x_sub_dx (map f p) = map f (x_sub_dx p) := by simp
```

Comment:
If the proof is `by simp`, then you should not mark the lemma with `@[simp]` (the linter will tell you that, too), because this new lemma does not make `simp` more powerful (it just makes it slower).
Also I think we don't need this lemma anyways, for the same reason.